src/validate.js

```diff
@@ -141,21 +141,28 @@ module.exports = validators => {
   };
 
   exports.customObjectError = val => {
-    let typeName = is.object(val) ? val.constructor.name : typeof val;
-
-    switch (typeName) {
-      case 'DocumentReference':
-      case 'FieldPath':
-      case 'FieldValue':
-      case 'GeoPoint':
-        return new Error(
-          `Detected an object of type "${typeName}" that doesn't match the expected instance. Please ensure that ` +
-            'the Firestore types you are using are from the same NPM package.'
-        );
-      default:
-        return new Error(
-          `Cannot use custom type "${typeName}" as a Firestore type.`
-        );
+    if (is.object(val) && val.constructor.name !== 'Object') {
+      switch (val.constructor.name) {
+        case 'DocumentReference':
+        case 'FieldPath':
+        case 'FieldValue':
+        case 'GeoPoint':
+          return new Error(
+            `Detected an object of type "${
+              val.constructor.name
+            }" that doesn't match the expected instance. Please ensure that ` +
+              'the Firestore types you are using are from the same NPM package.'
+          );
+        default:
+          return new Error(
+            "Firestore doesn't support serialization of JavaScript objects " +
+              "that were constructed using the 'new' operator."
```

Comment:
Interesting phrasing.  I'm not 100% sure if this will make sense to users or not.  I have two concerns:
1. "constructed using the 'new' operator" is kind of a weird differentiator for what works or not. I'm not sure if it'll resonate / makes sense for folks.
2. I wonder if anybody will get confused and think that `{ date: new Date(), location: new GeoPoint(...) }` might be disallowed since it uses new().

Is there a reason you removed the type name from the error?  I think that would address 2.  So I'm imagining an error something like:

> Couldn't serialize ${typeName} object.  Firestore doesn't support JavaScript objects with custom prototypes (i.e. that were created via the 'new' operator).

WDYT?  And can you make a similar change or open a bug to make the web client error match this?

Reply:
I removed the type name from the error message as the error got somewhat lengthy. That's probably not a good enough reason, so I put it back. I also mostly went with your re-wording in general.